### PR TITLE
[W5.7] [W17-1] Gu Wangfan

### PR DESCRIPTION
**Pull request by scalarmotion**

### CODE REVIEW

**Review by jhengy — COMMENTED** [review]
Good job. May want to add in Junit test for the new class you've created.

**Comment by scalarmotion** [comment]
JUnit tests were added in commit ce694c0c15271f065f07c6a3e87e14507f99e50e.

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -91,6 +91,23 @@ Returns `John Doe` but not `john`.
 * `find Betsy Tim John` +
 Returns Any person having names `Betsy`, `Tim`, or `John`.
 
+== Finding all persons whose phone numbers start with a prefix: `prefix`
```
Comment (amrut-prabhu) [inline docs/UserGuide.adoc:94]: Good work for updating the User Guide!

File: src/seedu/addressbook/commands/PhonePrefixCommand.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    /**
+     * Retrieves all persons in the address book whose phone numbers start with the given prefix.
+     *
+     * @param prefix for searching
+     * @return list of persons found
+     */
+    private List<ReadOnlyPerson> getPersonsWithPhonePrefix(String prefix) {
+        final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
+        for (ReadOnlyPerson person : addressBook.getAllPersons()) {
```
Comment (amrut-prabhu) [inline src/seedu/addressbook/commands/PhonePrefixCommand.java:47]: You can consider using Java 8 Streams here instead (though this is perfectly fine too)

File: test/expected.txt [hunk 3]
```diff
@@ -17,6 +17,9 @@
 || find: Finds all persons whose names contain any of the specified keywords (case-sensitive) and displays them as a list with index numbers.
 || Parameters: KEYWORD [MORE_KEYWORDS]...
 || Example: find alice bob charlie
+|| prefix: Finds all persons whose phone numbers start with the specified prefix and displays them as a list with index numbers.
```
Comment (amrut-prabhu) [inline test/expected.txt:20]: Good work updating the I/O tests!

File: test/java/seedu/addressbook/commands/PhonePrefixCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        import seedu.addressbook.data.person.Address;
+        import seedu.addressbook.util.TypicalPersons;
+
+public class PhonePrefixCommandTest {
+
+    private final AddressBook addressBook = new TypicalPersons().getTypicalAddressBook();
+    private final TypicalPersons td = new TypicalPersons();
+
+    @Test
+    public void execute() throws IllegalValueException {
```
Comment (amrut-prabhu) [inline test/java/seedu/addressbook/commands/PhonePrefixCommandTest.java:27]: Good work on the JUnit testing!

File: test/java/seedu/addressbook/commands/PhonePrefixCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
+package seedu.addressbook.commands;
+
+        import static org.junit.Assert.assertEquals;
```
Comment (amrut-prabhu) [inline test/java/seedu/addressbook/commands/PhonePrefixCommandTest.java:3]: None of these imports should be indented

**Review by amrut-prabhu — APPROVED** [review]
Good work on the enhancement and LO @scalarmotion !

Please close the PR after reading my comments.

**Pull request closed without merging**